### PR TITLE
Adding e2e test framework

### PR DESCRIPTION
**Pull request by caesarxuchao**
Adding the e2e test framework. The test workflow is:
1. create objects
2. restart apiserver, with --storage-verions setting a different storage version
3. run migrator
4. using etcdctl checking if the objects are migrated to the right version.

In this pull, I set up the e2e test framework but didn't really run the migrator. I'll run the migrator in a future pull.

### CODE REVIEW

File: test/e2e/test-cmd.sh [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+# Sanity check.
+# Note that the log inidicates that the kubectl in the test driver is v1.10.7
+kubectl version
+
+# TODO: in the real test, create objects here.
+
+# Change the apiserver manifest to change the storage version.
+gcloud compute --project "${PROJECT}" ssh --zone "${KUBE_GCE_ZONE}" "${CLUSTER_NAME}-master" --command \
+"sudo sed -i \"s/--v=/--storage-versions=apps\/v1beta2 --v=/\" /etc/kubernetes/manifests/kube-apiserver.manifest"
```
Comment (krzyzacy) [inline test/e2e/test-cmd.sh]: don't hard-code v1beta2?

File: test/e2e/test-cmd.sh [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+result=$(gcloud compute --project "${PROJECT}" ssh --zone "${KUBE_GCE_ZONE}" "${CLUSTER_NAME}-master" --command \
+  "docker ps")
+etcd_container=$(echo "${result}" | grep "etcd-server-${CLUSTER_NAME}-master" | grep -v pause | cut -d ' ' -f 1)
+
+version=$(gcloud compute --project "${PROJECT}" ssh --zone "${KUBE_GCE_ZONE}" "${CLUSTER_NAME}-master" --command \
+  "docker exec ${etcd_container} /bin/sh -c \"ETCDCTL_API=3 etcdctl get /registry/controllerrevisions/default/sample\" | grep apps")
+# Remove the trailing non-printable character. The data is encoded in proto, so
+# it has non-printable characters.
+version=$(tr -dc '[[:print:]]' <<< "${version}")
+echo ${version}
```
Comment (krzyzacy) [inline test/e2e/test-cmd.sh]: nit: quote the bash variable, also line 76

Comment (caesarxuchao) [reply test/e2e/test-cmd.sh]: Done in both.

FYI, it's not necessary to use quotes on the left operand in the double square. https://unix.stackexchange.com/questions/32210/why-does-parameter-expansion-with-spaces-without-quotes-work-inside-double-brack

File: test/e2e/test-cmd.sh (hunk 1 above)
Comment (caesarxuchao) [reply test/e2e/test-cmd.sh]: I will lock the apiserver version in the job config. That should make this hardcoding reliable.

File: test/e2e/test-cmd.sh (hunk 2 above)
Comment (krzyzacy) [reply test/e2e/test-cmd.sh]: ack :-) I'm not expert on bash :jack_o_lantern:

**Pull request merged**